vault/resource_identity_oidc_client.go

```diff
@@ -0,0 +1,156 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+const identityOIDCClientPathPrefix = "identity/oidc/client"
+
+func identityOIDCClientResource() *schema.Resource {
+	return &schema.Resource{
+		Create: identityOIDCClientCreateUpdate,
+		Update: identityOIDCClientCreateUpdate,
+		Read:   identityOIDCClientRead,
+		Delete: identityOIDCClientDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Description: "The name of the client.",
+				Required:    true,
+			},
+			"key": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
```

Comment:
Nice! I think ForceNew is used for immutable fields? (trying to learn TF here :))

Reply:
@austingebauer exactly! The TF SDK will let the user know that they are trying to change an immutable field, which would result in the destruction of the previously configured resource and creation of a new Client resource :)